qt/widgets/plugins/algorithm_dialogs/src/PeriodicTableWidget.cpp

```diff
@@ -45,97 +45,95 @@ void PeriodicTableWidget::ColourElements() {
 void PeriodicTableWidget::ColourActinides(
     const QVector<QPushButton *> &actinides) {
   QString buttonColourStr = "background-color: rgb(255, 85, 127, 255)";
-  for (auto i = actinides.begin(); i != actinides.end(); i++) {
-    ColourButton(*i, buttonColourStr);
+  for (auto actinide : actinides) {
+    ColourButton(actinide, buttonColourStr);
     update();
   }
 }
 void PeriodicTableWidget::ColourAlkaliMetals(
     const QVector<QPushButton *> &alkaliMetals) {
   QString buttonColourStr = "background-color: rgb(255, 255, 0, 255)";
-  for (auto i = alkaliMetals.begin(); i != alkaliMetals.end(); i++) {
-    ColourButton(*i, buttonColourStr);
+  for (auto alkaliMetal : alkaliMetals) {
+    ColourButton(alkaliMetal, buttonColourStr);
     update();
   }
 }
 void PeriodicTableWidget::ColourAlkalineEarthMetals(
     const QVector<QPushButton *> &alkalineEarthMetals) {
   QString buttonColourStr = "background-color: rgb(170, 170, 127, 255)";
-  for (auto i = alkalineEarthMetals.begin(); i != alkalineEarthMetals.end();
-       i++) {
-    ColourButton(*i, buttonColourStr);
+  for (auto alkalineEarthMetal : alkalineEarthMetals) {
+    ColourButton(alkalineEarthMetal, buttonColourStr);
     update();
   }
 }
 void PeriodicTableWidget::ColourHalogens(
     const QVector<QPushButton *> &halogens) {
   QString buttonColourStr = "background-color: rgb(0, 255, 255, 255)";
-  for (auto i = halogens.begin(); i != halogens.end(); i++) {
-    ColourButton(*i, buttonColourStr);
+  for (auto halogen : halogens) {
+    ColourButton(halogen, buttonColourStr);
     update();
   }
 }
 void PeriodicTableWidget::ColourLanthanides(
     const QVector<QPushButton *> &lanthanides) {
   QString buttonColourStr = "background-color: rgb(170, 85, 255, 255)";
-  for (auto i = lanthanides.begin(); i != lanthanides.end(); i++) {
-    ColourButton(*i, buttonColourStr);
+  for (auto lanthanide : lanthanides) {
+    ColourButton(lanthanide, buttonColourStr);
     update();
   }
 }
 void PeriodicTableWidget::ColourMetalloids(
     const QVector<QPushButton *> &metalloids) {
   QString buttonColourStr = "background-color: rgb(255, 170, 255, 255)";
-  for (auto i = metalloids.begin(); i != metalloids.end(); i++) {
-    ColourButton(*i, buttonColourStr);
+  for (auto metalloid : metalloids) {
+    ColourButton(metalloid, buttonColourStr);
     update();
   }
 }
 void PeriodicTableWidget::ColourNobleGases(
     const QVector<QPushButton *> &nobleGases) {
   QString buttonColourStr = "background-color: rgb(255, 170, 0, 255)";
-  for (auto i = nobleGases.begin(); i != nobleGases.end(); i++) {
-    ColourButton(*i, buttonColourStr);
+  for (auto nobleGase : nobleGases) {
```

Comment:
`nobleGas`?

Reply:
Ah missed this one- clang-tidy created a variable name by removing the "s" at the end if there is one. Sometimes you get non-words like gase. I'll fix this

Reply:
Thanks. I'm generally very impressed at `clang-tidy`s naming choices like deciphering `index` from indices`.